### ISSUE TITLE
Files in `root_dir` not watched properly when using `include_dir`

### ISSUE THREAD

**Issue by tprestegard** [cosmtrek/air#496]
air version: 1.49.0

I would like to be able to watch all files with specific extensions in a set of directories as well as some files in `root_dir`. I have an example module layout like this:

```
$ tree
.
├── air.toml
├── go.mod
├── main.go
├── mypkg
|   └── exec.go
└── otherpkg
    └── file.go
```

For example, I would like to watch all `.go` files in `mypkg`, `main.go`, and `go.mod`. I first tried the following config:
```
include_dir = ["mypkg"]
include_ext = ["go"]
include_file = ["go.mod"]
```

I ran this in Docker with the following command:
```
docker run --rm -v $PWD:/src -w /src cosmtrek/air:v1.49.0 -d -c /src/air.toml
```

With this config, changes to `mypkg/exec.go` were detected fine and triggered a rebuild, no changes to `main.go` were detected, and some changes to `go.mod` were detected, but it did not trigger a rebuild. For `go.mod` it seems to not process all the same file events as for `mypkg/exec.go`.

Specifically, the *first* save to `go.mod` produces these file events:
```
[17:09:38] event: REMOVE        "/src/go.mod"                                                                                                                                                              
[17:09:38] event: CHMOD         "/src/go.mod"
```
and successive changes to `go/mod` do not produce *any* file events.

For `mypkg/exec.go`, many more file events are detected when the file is saved, and they are observed on every save:
```
[17:16:16] event: RENAME        "/src/mypkg/exec.go"
[17:16:16] event: WRITE         "/src/mypkg/exec.go"
[17:16:16] event: CREATE        "/src/mypkg/exec.go"
[17:16:16] mypkg/exec.go has changed
[17:16:16] mypkg/exec.go has changed
[17:16:16] event: CHMOD         "/src/mypkg/exec.go"
[17:16:17] flushing events
[17:16:17] mypkg/exec.go has changed
```

So it seems like I could get `main.go` to be watched by adding it to `include_file`. But the watching provided by this configuration seems incomplete as file events are not always detected and do not trigger rebuilds, as seen for the example with `go.mod` above. I could also remove the `include_dir` setting and then things mostly work as desired, but then it also watches the `otherpkg` directory, which is not desired. Maybe I could do that and then used `exclude_dir = ["otherpkg"]`? That seems to work, but it doesn't scale well when there are lots of directories and you really only want to include one or a few of them. I also adding `"."` to `include_dir`, which seems to have no effect.

Maybe this is a misunderstanding on my part of how the various parameters for including and excluding things are supposed to be used. For example, maybe I need to include `"mod"` in the `include_ext` to have `go.mod` watched properly? It's not clear (I did try it and it didn't help). These could *really* use some documentation because it's not clear at all how they interact, which take precedence, which can work together, etc.

To summarize:
* There is some strange behavior with file watching when using `include_file` in combination with `include_dir` where incomplete sets of file events are generated and the binary rebuild is not triggered.
* The interactions between the various `include_*` and `exclude_*` parameters are confusing and could be better documented.

**Comment by silverwind** [comment]
Also noticed this, minimal repro should be:

```toml
include_dir = ["api"]
include_file = [".env"]
```

Editing `.env` never restarts the server.

**Comment by xiantang** [comment]
> Also noticed this, minimal repro should be:
> 
> include_dir = ["api"]
> include_file = [".env"]
> Editing `.env` never restarts the server.

The reason is `filenotify` only watches for inode, so if u are using an atomic saving editor like vim, the inode would be changed

**Comment by silverwind** [comment]
> only watches for inode

I find that hard to believe, but if true, I think file watching should use a content-hash based mechanism.

Atomic save feature is when editors write into a temp file and move it in place, in which case inode should always change. Non-atomic save may be inode-preserving.

In any case, I think inode it too brittle, just hash the file's content.

**Comment by xiantang** [comment]
I already raised a PR to fix this https://github.com/air-verse/air/pull/908. I don't really want to change too much. This issue only happened on include_file. if filenotify watches for folders, it works well.

> > only watches for inode
> 
> I find that hard to believe, but if true, I think file watching should use a content-hash based mechanism.
> 
> Atomic save feature is when editors write into a temp file and move it in place, in which case inode should always change. Non-atomic save may be inode-preserving.
> 
> In any case, I think inode it too brittle, just hash the file's content.

**Comment by silverwind** [comment]
Minimal fix sounds good, but please ensure your premise regarding atomic save is actually correct.

**Comment by xiantang** [comment]
<img width="1664" height="734" alt="Image" src="https://github.com/user-attachments/assets/3231814a-1129-4c80-9683-1125b2ea2022" />
And I found sth really interesting, I just `:wq` for both go.mod and cmd/app/main.go, but the events are not the same.

**Comment by xiantang** [comment]
> Minimal fix sounds good, but please ensure your premise regarding atomic save is actually correct.

i tested https://github.com/air-verse/air/pull/908 and looks good, if u got time pls help me test it as well, thanks in advance.
if everything's ok, i will merge to master tmr.
